sdl_android_lib/src/com/smartdevicelink/transport/SdlRouterService.java

```diff
@@ -328,6 +328,9 @@ public RouterHandler(SdlRouterService provider){
 	    	
 	        @Override
 	        public void handleMessage(Message msg) {
+	        	if(this.provider == null || this.provider.get() == null){
```

Comment:
Could make this.provider variable final, initialize it once in constructor and not to check for (this.provider == null) each time after. Only this.provider.get() can actually be null.